src/stamp/modeling/config.py

```diff
@@ -27,20 +27,24 @@ class TrainConfig(BaseModel):
     filename_label: PandasLabel = "FILENAME"
 
     # Dataset and -loader parameters
-    bag_size: int = 512
-    num_workers: int = min(os.cpu_count() or 1, 16)
+    bag_size: int = 1024
+    num_workers: int = min(os.cpu_count() or 1, 8)
 
     # Training paramenters
     batch_size: int = 64
     max_epochs: int = 64
     patience: int = 16
     accelerator: str = "gpu" if torch.cuda.is_available() else "cpu"
-
+    lr: float = 1e-4
+    freeze_base: bool = True
+    freeze_cobra: bool = False
+    
     # Experimental features
     use_vary_precision_transform: bool = False
     use_alibi: bool = False
+    use_cobra: bool = False
```

Comment:
Are these options (freeze_base, freeze_cobra, use_cobra) all COBRA-specific? If so, you could specify a CobraConfig with COBRA-specific configs, e.g.:
```python
class CobraConfig(BaseModel):
    enabled: bool = False
    freeze_base: bool = True
    ...
```

And then set an attribute `cobra: CobraConfig` in `TrainConfig`. This would make it clearer that certain configuration options, e.g. freeze_base, are only applicable if COBRA is enabled.